avocado/utils/podman.py

```diff
@@ -0,0 +1,102 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 Red Hat Inc.
+# Authors : Beraldo Leal <bleal@redhat.com>
+
+"""
+This module provides an basic API for interacting with podman.
+
+This module it was designed to be executed in async mode. Remember this when
+consuming this API.
+"""
+
+import json
+import logging
+from asyncio import create_subprocess_exec, subprocess
+from shutil import which
+
+LOG = logging.getLogger(__name__)
+
+
+class PodmanException(Exception):
+    pass
+
+
+class Podman:
+    def __init__(self, podman_bin=None):
+        path = which(podman_bin or 'podman')
+        if not path:
+            msg = f"Podman binary {podman_bin} is not available on the system."
+            raise PodmanException(msg)
+
+        self.podman_bin = path
+
+    async def execute(self, *args):
+        """Execute a command and return the returncode, stdout and stderr.
+
+        :param *args: Variable length argument list to be used as argument
+                      during execution.
+        :rtype: tuple with returncode, stdout and stderr.
+        """
+        try:
+            proc = await create_subprocess_exec(self.podman_bin,
+                                                *args,
+                                                stdout=subprocess.PIPE,
+                                                stderr=subprocess.PIPE)
+            stdout, stderr = await proc.communicate()
+        except (FileNotFoundError, PermissionError) as ex:
+            # Since this method is also used by other methods, let's
+            # log here as well.
+            msg = "Could not execute the command."
+            LOG.error("%s: %s", msg, str(ex))
+            raise PodmanException(msg) from ex
+
+        if proc.returncode != 0:
+            msg = f"Could not execute the command: {proc.returncode}:{stderr}."
+            LOG.error(msg)
+            raise PodmanException(msg)
+
+        return proc.returncode, stdout, stderr
+
+    async def get_python_version(self, image):
+        """Return the current Python version installed in an image.
+
+        :param str image: Image name. i.e: 'fedora:33'.
+        :rtype: tuple with both: major, minor numbers and executable path.
+        """
+
+        entrypoint = json.dumps(["/usr/bin/env", "python3", "-c",
+                                 ("import sys; print(sys.version_info.major, "
+                                  "sys.version_info.minor, sys.executable)")])
+
+        try:
+            _, stdout, _ = await self.execute("run",
+                                              "--rm",
+                                              f"--entrypoint={entrypoint}",
+                                              image)
+        except PodmanException as ex:
+            raise PodmanException("Failed getting Python version.") from ex
+
+        if stdout:
+            return tuple(stdout.decode().strip().split())
```

Comment:
I did a tweak to the test:

```python
    def test_python_version(self):
        """
        :avocado: requirement={"type": "package", "name": "podman", "action": "check"}
        :avocado: tags=slow
        """
        podman = Podman()
        loop = asyncio.get_event_loop()
        coro = podman.get_python_version('fedora:34')
        result = loop.run_until_complete(coro)
        self.assertEqual(result, (3, 9, '/usr/bin/python3'))
```

and IMO this should return a tuple of (int, int, str)